### PR TITLE
Repo cleanups

### PR DESCRIPTION
**Pull request by MikeHathaway**
**Updates**
- Add tests for Maths.sol
- Expand test coverage for grant fund branches
- Move `GrantFundTestHelper` into test/utils directory
- Remove TODOs
- Expand natspec comments
- Move natspec into relevant interfaces
- Remove `payable` modifiers from execute methods

**Remaining Issues**

- Reorganize sections in the GrantFunding abstract contracts away from `Proposal Functions` `Voting Functions` etc.

### CODE REVIEW

File: test/ExtraordinaryFunding.t.sol [hunk 1]
```diff
@@ -297,7 +296,8 @@ contract ExtraordinaryFundingGrantFundTest is GrantFundTestHelper {
     }
 
     function testProposeExtraordinaryMultipleCalldata() external {
-        // TODO: finish implementing this test
+        // TODO: finish implementing this test -> check handling the submission of proposals that have multiple calldatas in the calldata array
```
Comment (prat-gpt) [inline test/ExtraordinaryFunding.t.sol]: Are we keeping these TODOs?

**Review by prat-gpt — APPROVED** [review]
LGTM! Added a comment regarding TODOs.

File: test/ExtraordinaryFunding.t.sol (hunk 1 above)
Comment (MikeHathaway) [reply test/ExtraordinaryFunding.t.sol]: Removed

File: src/grants/interfaces/IExtraordinaryFunding.sol [hunk 2]
```diff
@@ -22,20 +22,79 @@ interface IExtraordinaryFunding {
...
+        uint256[] memory values_,
+        bytes[] memory calldatas_,
+        string memory description_) external returns (uint256 proposalId_);
+
+    /**********************/
+    /*** View Functions ***/
+    /**********************/
+
+    /**
+     *  @notice Mapping of proposalIds to {ExtraordinaryFundingProposal} structs.
```
Comment (ith-harvey) [inline src/grants/interfaces/IExtraordinaryFunding.sol:82]: NIT: remove brackets

File: src/grants/interfaces/IStandardFunding.sol [hunk 3]
```diff
@@ -115,4 +115,138 @@ interface IStandardFunding {
...
+    /**
+     * @notice Check if a slate of proposals meets requirements, and maximizes votes. If so, update QuarterlyDistribution.
+     * @param  proposalIds_ Array of proposal Ids to check.
+     * @param  distributionId_ Id of the current quarterly distribution.
+     * @return Boolean indicating whether the new proposal slate was set as the new top slate for distribution.
+     */
+    function checkSlate(uint256[] calldata proposalIds_, uint256 distributionId_) external returns (bool);
+
+    /**
+     * @notice distributes delegate reward based on delegatee Vote share.
```
Comment (ith-harvey) [inline src/grants/interfaces/IStandardFunding.sol:131]: You were aligning above and below comments, do we need to align?

File: src/grants/interfaces/IExtraordinaryFunding.sol (hunk 2 above)
Comment (MikeHathaway) [reply src/grants/interfaces/IExtraordinaryFunding.sol:82]: brackets are consistent with how we do it in pool repo

File: src/grants/interfaces/IStandardFunding.sol (hunk 3 above)
Comment (MikeHathaway) [reply src/grants/interfaces/IStandardFunding.sol:131]: good catch will align

File: src/grants/interfaces/IStandardFunding.sol [hunk 4]
```diff
@@ -115,4 +115,138 @@ interface IStandardFunding {
...
+    /**
+     * @notice Retrieve the current QuarterlyDistribution distributionId.
+     * @return The current distributionId.
+     */
+    function getDistributionId() external view returns (uint256);
+
+    /**
+     * @notice Mapping of distributionId to {QuarterlyDistribution} struct.
+     * @param  distributionId_ The distributionId to retrieve the QuarterlyDistribution struct for.
+     * @return                 The retrieved struct's distributionId.
```
Comment (ith-harvey) [inline src/grants/interfaces/IStandardFunding.sol]: I understand these are off of a struct but I think we should list the property / give each of these returns a name.. even though it is technically an unnamed variable. Thoughts?

**Review by ith-harvey — COMMENTED** [review]
LGTM, Left some comments about the comments which I think are worth consideration.

**Pull request merged**